odbc_fdw.c

```diff
@@ -364,136 +670,68 @@ odbcGetOptions(Oid foreigntableid, char **svr_dsn, char **svr_driver, char **svr
 		elog(DEBUG1, "odbcGetOptions");
 	#endif
 
-	/*
-	 * Extract options from FDW objects.
-	 */
-	table = GetForeignTable(foreigntableid);
-	server = GetForeignServer(table->serverid);
-	mapping = GetUserMapping(GetUserId(), table->serverid);
+    server  = GetForeignServer(server_oid);
+	mapping = GetUserMapping(GetUserId(), server_oid);
 
 	options = NIL;
-	options = list_concat(options, table->options);
+	options = list_concat(options, add_options);
 	options = list_concat(options, server->options);
 	options = list_concat(options, mapping->options);
 
-	*mapping_list = NIL;
-	/* Loop through the options, and get the foreign table options */
-	foreach(lc, options)
-	{
-		DefElem *def = (DefElem *) lfirst(lc);
-
-		if (strcmp(def->defname, "dsn") == 0)
-		{
-			*svr_dsn = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "driver") == 0)
-		{
-			*svr_driver = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "host") == 0)
-		{
-			*svr_host = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "port") == 0)
-		{
-			*svr_port = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "database") == 0)
-		{
-			*svr_database = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "schema") == 0)
-		{
-			*svr_schema = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "table") == 0)
-		{
-			*svr_table = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "sql_query") == 0)
-		{
-			*sql_query = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "sql_count") == 0)
-		{
-			*sql_count = defGetString(def);
-			continue;
-		}
-
-		if (strcmp(def->defname, "username") == 0)
-		{
-			*username = defGetString(def);
-			continue;
-		}
+	extract_odbcFdwOptions(options, extracted_options);
+}
 
-		if (strcmp(def->defname, "password") == 0)
-		{
-			*password = defGetString(def);
-			continue;
-		}
+/*
+ * Fetch the options for a odbc_fdw foreign table.
+ */
+static void
+odbcGetTableOptions(Oid foreigntableid, odbcFdwOptions *extracted_options)
+{
+	ForeignTable    *table;
+	ForeignServer   *server;
 
-		/* Column mapping goes here */
-		*mapping_list = lappend(*mapping_list, def);
-	}
+	#ifdef DEBUG
+		elog(DEBUG1, "odbcGetTableOptions");
```

Comment:
What I meant is why do you use `DEBUG` in some places and `DEBUG1` in some other places. What's the criteria you're using? Otherwise I'd opt for simplifying the code and using just `DEBUG`.